backend/resources/books.py

```diff
@@ -124,14 +124,15 @@ def get(self):
         parser.add_argument('score', type=bool, required=False,
                             help="Indicates if returning the score of the book is needed .")
         data = parser.parse_args()
+        books = BooksModel.query.filter_by(vendible=True)
         with lock:
```

Comment:
Las query las has de fer dintre del lock, que sino es poden produir errors si algu elimina un llibre durant la query.
```suggestion
        with lock:
            books = BooksModel.query.filter_by(vendible=True)
```